packages/ws-client/src/data-source.ts

```diff
@@ -160,3 +162,29 @@ export function createWSDataSource(wsUrl: string): DataSource {
 
   return dataSource
 }
+
+const createInfoStream = (rws: ReconnectingWebSocket) => {
+  const [writeInfo, info] = createWritableStream({ status: readyStateToStatus(rws.readyState) })
+
+  const updateStatus = () => writeInfo({ status: readyStateToStatus(rws.readyState) })
+  rws.addEventListener('open', updateStatus)
+  rws.addEventListener('close', updateStatus)
+  rws.addEventListener('error', updateStatus)
+
+  return info
+}
+
+const readyStateToStatus = (readyState: number) => {
```

Comment:
RWS does not report "connecting" and "closing" statuses, but they are valid as per spec. I think they may happen inside `onError()` event. I always read the status from `rws.readyState` and convert to one of these.

Reply:
Ok nice